internal/clients/vault_secrets.go

```diff
@@ -118,19 +124,33 @@ func CreateVaultSecretsAppSecret(ctx context.Context, client *Client, loc *share
 
 // OpenVaultSecretsAppSecret will retrieve the latest secret for a Vault Secrets app, including it's value.
 func OpenVaultSecretsAppSecret(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, appName, secretName string) (*secretmodels.Secrets20230613OpenSecret, error) {
-
 	getParams := secret_service.NewOpenAppSecretParams()
 	getParams.Context = ctx
 	getParams.AppName = appName
 	getParams.SecretName = secretName
 	getParams.LocationOrganizationID = loc.OrganizationID
 	getParams.LocationProjectID = loc.ProjectID
 
-	getResp, err := client.VaultSecrets.OpenAppSecret(getParams, nil)
-	if err != nil {
-		return nil, err
+	var getResp *secret_service.OpenAppSecretOK
+	var err error
+	for attempt := 0; attempt < retryCount; attempt++ {
+		getResp, err = client.VaultSecrets.OpenAppSecret(getParams, nil)
+		if err != nil {
```

Comment:
Is it possible to build a generic module that retries with the right backoff based on status code so that others can also use it?

Reply:
generic wrapper is doable but is probably a little bigger undertaking at this point. let me revisit it after this so that the issue highlighted is fixed.

Reply:
Thanks. This can be implemented later in subsequent PR.